src/Umbraco.Core/Models/Language.cs

```diff
@@ -20,11 +18,17 @@ public class Language : EntityBase, ILanguage
         private bool _mandatory;
         private int? _fallbackLanguageId;
 
-        public Language(string isoCode)
+        public Language(string isoCode, string cultureName)
         {
             IsoCode = isoCode;
+            CultureName = cultureName;
         }
 
+        [Obsolete("Use the constructor also specifying the culture name instead.")]
+        public Language(string isoCode)
+            : this(isoCode, isoCode)
+        { }
```

Comment:
Let's use the simplest fallback by making the `CultureName` the same as the `IsoCode` when it's not specified.